### ISSUE TITLE
(Docker) deleting bind cleares external storage

### ISSUE THREAD

**Issue by mirisbowring** [nextcloud/docker#1493]
Hi,
I am not exactly sure whether this is a bug or a designissue.
I am trying to describe the problem.

### Steps to reproduce
1. Use default `docker-compose.yml` from nextcloud dockerhub to deploy a instance
2. install / enable external storage app
3. mount external drive to `/mnt/abcdef`
4. bind `/my/custom/path`to `/var/www/html`(this is proposed in the compose file from dockerhub)
5. bind `/mnt/abcdef` to `/my-external-storage` in the volume section of the compose file
6. configure the external storage in the nextcloud ui (it's available from the container at `/my-external-storage`)
7. Use storage ...

Here the trouble began:

8. Shutdown Stack
9. backup `/my/custom/path`(`/var/www/html` directory from container)
10. wanted a "clean" install -> deleted the `/my/custom/path` (`/var/www/html`directory from container)
11. start new stack (same compose file)
12. configure external storage
13. wonder why no data shown
14. shutdown stack -> restore backed up app-folder -> start stack
15. still no data
16. check if drive is still mounted
17. see `df -h` that the drive is empty :scream:
18. check backed up folder
19. see that external storage folder is there  

### Expected behaviour
Data should not be deleted if i delete the App-Folder (/var/www/html) because it is on a completely different path.

### Actual behaviour
Somehow, the data got deleted. Is it Possible that there is a Hard- or Softlink to the mounted drive? So deleting the link, deletes the mounted folder?

### Server configuration

**Operating system:** Ubuntu Server 20.04 with latest Docker

**Web server:** Apache Container

**Database:** MariaDB

**PHP version:** as delivered in the container

**Nextcloud version:** 21.0.1

**Updated from an older Nextcloud/ownCloud or fresh install:**

**Where did you install Nextcloud from:** https://hub.docker.com/_/nextcloud

**Are you using external storage, if yes which one:** local

**Are you using encryption:** no

### Client configuration
**Browser:** Chrome

**Operating system:** MacOS / Windows / Linux

**Comment by mirisbowring** [comment]
I hate nextcloud.
I just pulled a new Image and am getting the following log output:

```
Configuring Redis as session handler
Initializing nextcloud 21.0.2.1 ...
Upgrading nextcloud from 21.0.1.1 ...
rsync: delete_file: rmdir(musiclib) failed: Device or resource busy (16)
rsync: delete_file: rmdir(moviedb1) failed: Device or resource busy (16)
rsync error: some files/attrs were not transferred (see previous errors) (code 23) at main.c(1207) [sender=3.1.3]
```

musiclib and moviedb1 are mounted hard drives that are mapped as external storage to nextcloud.
EVERYTHING on both drives is wiped! F*ck it!

16TB Data gone for no reason... How can this happen.

Why the F*ck are external storages deleted when migrating to a new version?

**Comment by mirisbowring** [comment]
This Error also occures for upgrading to 21.0.3:
I learned fromthis behaviour and mounted the drive as Read Only so that rsync is not able to purge my files:
<img width="319" alt="Screenshot 2021-07-04 at 14 47 34" src="https://user-images.githubusercontent.com/15715080/124385690-257d9a80-dcd7-11eb-93ee-4b92c6be2a13.png">

After this part, there are thousands of log entries that rsync is not able to delete a file (it is trying to delete every single file one by one on that mounted external storage).

**Comment by joshtrichards** [comment]
Duplicate of #1528
Fixed on #1862
